python/cuxfilter/charts/core/aggregate/core_aggregate.py

```diff
@@ -82,7 +82,9 @@ def __init__(
         """
         self.x = x
         self.y = y
-        self.input_stride = step_size
+        if step_size or data_points:
+            self.custom_binning = True
```

Comment:
Why not a computed read-only property for `custom_binning` so that it does not have to be manually put in sync

Reply:
problem is `custom_binning` only needs to be set when chart is declared, and it should be static throughout. Computed would make sense when `custom_binning` is expected to change throughout the lifetime of the dashboard, right?

Now its only updated when chart object is created (cuxfilter.charts.bar/line). the previous logic updated it whenever dashboard was initialized.

Reply:
only if one of input `data_points` or `step_size` exist, will `custom_binning` be True. `sefl.data_points` and `self.stride` are updated when we initialize the dashbooard (chart.initiate_chart is called), and `custom_binning` can get out of sync if we run those cells twice.

Reply:
>  and it should be static throughout. 

I would say that is another point in favor of a getter-only property -- it can't accidentally / unintentionally set

>  and custom_binning can get out of sync if we run those cells twice.

I guess I don't understand the full codepath. You are saying it is normal operation for there to be situations where `custom_binning` is True, but yet `data_points` and `stride` become un-set?

Reply:
okay, I get your point now, I somehow missed the `read-only` part,  `read-only getter` does make sense. I'll update it.

Reply:
let me know if this looks good, `custom_binning` is now a read-only computed property, `self._stride` and `self._data_points` are internal variables, which store input stride and input data_points, based on which `self.custom_binning` is computed.

`self.stride` and `self.data_points` are updated based on the `calculate_source` fn, and are updated whenever chart source is calculated (used to calculate data_tiles and range_slider step size)